src/qz/utils/PrintingUtilities.java

```diff
@@ -219,4 +229,37 @@ public static void processPrintRequest(Session session, String UID, JSONObject p
         }
     }
 
+    public static void cancelJobs(Session session, String UID, JSONObject params) {
+        try {
+            NativePrinter printer = PrintServiceMatcher.matchPrinter(params.getString("printerName"));
+            if (SystemUtilities.isWindows()) {
+                WinNT.HANDLEByReference phPrinter = new WinNT.HANDLEByReference();
+                WinspoolEx.INSTANCE.OpenPrinter(printer.getName(), phPrinter, null);
+                Winspool.JOB_INFO_1[] jobs =  WinspoolUtil.getJobInfo1(phPrinter);
+                // skip retained jobs and complete jobs
+                int skipMask = (int)WmiJobStatusMap.RETAINED.getRawCode() | (int)WmiJobStatusMap.PRINTED.getRawCode();
+                int deletedCount = 0;
+                for (Winspool.JOB_INFO_1 job : jobs) {
+                    if ((job.Status & skipMask) != 0) continue;
+                    boolean result = WinspoolEx.INSTANCE.SetJob(phPrinter.getValue(), job.JobId, 0, null, WinspoolEx.JOB_CONTROL_DELETE);
+                    if (result) {
+                        deletedCount++;
+                    } else {
+                        log.warn("Job deletion error for job#{}, error code:{}", job.JobId, Kernel32.INSTANCE.GetLastError());
+                    }
+                }
+                log.info("Deleting {} jobs", deletedCount);
+            } else {
+                ArrayList<Integer> jobIds = CupsUtils.listJobs(printer.getPrinterId());
```

Comment:
Does this need macNameFix like https://github.com/qzind/tray/blob/2605e2a8b7c03159f173a6489154a106e30c122e/src/qz/printer/status/StatusMonitor.java#L94?

Reply:
No, AFAIR.

It should be documented better, but `getPrinterId` is the "ugly" CUPS id of the printer (e.g. "Zebra_LP2844"), so this is a safe operation, not unlike our raw printing API here:

https://github.com/qzind/tray/blob/2605e2a8b7c03159f173a6489154a106e30c122e/src/qz/printer/status/CupsUtils.java#L57

The reason the helper isn't needed is because `NativePrinter` is "namefix" aware, as it's constructed with two values:
* name: What Java calls this printer
* printerId: What the OS calls this printer

In Windows and Linux, these values are always identical, but they're different on Mac, so we're unknowingly appling the name fix by using the `NativePrinter` object.

On the other hand, `StatusMonitor` is accepting raw `String` values as input, so they must be sanitized for Mac.